x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  Logger,
+  ElasticsearchClient,
+  SavedObjectsFindResult,
+  SavedObjectsClientContract,
+} from 'kibana/server';
+import { BackgroundSessionStatus, BackgroundSessionSavedObjectAttributes } from '../../../common';
+import { BACKGROUND_SESSION_TYPE } from '../../saved_objects';
+import { getSearchStatus } from './get_search_status';
+import { getSessionStatus } from './get_session_status';
+
+export async function checkRunningSessions(
+  savedObjectsClient: SavedObjectsClientContract,
```

Comment:
Any reason why this uses a savedObjectsClient and not a SessionClient object?

Reply:
Yes! Because the sessionClient is scoped to a request, and the monitoring task is not.